src/meshweb/views.py

```diff
@@ -33,3 +37,40 @@ def index(request: HttpRequest) -> HttpResponse:
     }
     context = {"links": links}
     return HttpResponse(template.render(context, request))
+
+
+def maintenance(request: HttpRequest) -> HttpResponse:
+    if not flag_enabled("MAINTENANCE_MODE"):
+        return HttpResponseRedirect(reverse("main"))
+    template = loader.get_template("meshweb/maintenance.html")
+    context = {
+        "message": "Please check back later.",
+        "redirect": "",
+    }
+    return HttpResponse(template.render(context, request))
+
+
+@api_view(["POST"])
+@permission_classes([IsSuperUser])
```

Comment:
```suggestion
@permission_classes([HasMaintenanceModePermission])
```
Doing it this way will have more runtime flexibility. I don't like coupling our application logic to the Django concept of superuser like that